### PR TITLE
[WIP] Incorporate Skinning into SIRo

### PR DESCRIPTION
**Pull request by xavierpuigf**
## Motivation and Context

Adds skinned humanoids into SIRo

## How Has This Been Tested

Training and Eval of humanoid models work

## Types of changes

- **\[Docs change\]** Addition or changes to the documentation
- **\[Experiment\]** A pull request that add new features to the [habitat-baselines](/habitat-baselines/) training codebase. Experiments Pull Requests can be any size, must have smoke/integration tests and be isolated from the rest of the code. Your code additions should not rely on other habitat-baselines code. This is to avoid dependencies between different parts of habitat-baselines. You must also include a README that will document how to use your new feature or trainer. **You** will be the maintainer of this code, if the code becomes stale or is not supported often enough, we will eventually remove it.

## Checklist

- [X] My code follows the code style of this project.
- [X] I have updated the documentation if required.
- [X] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [X] I have completed my CLA (see **CONTRIBUTING**)
- [X] I have added tests to cover my changes if required.

### CODE REVIEW

File: SIRO_README.md [hunk 1]
```diff
@@ -14,8 +14,7 @@ Project-specific README for SIRo.
     * Keep an eye on relevant [commits to main](https://github.com/facebookresearch/habitat-sim/commits/main) to help you decide when to update/rebuild Habitat-sim.
     * If you encounter the issue about cmake when building from source using Mac `missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun`, you can solve this by first install the tools `xcode-select --install`.
 1. Download humanoid data.
-    * From the Habitat-lab root directory, `python -m habitat_sim.utils.datasets_download --uids humanoid_data  --data-path data/`
-    * Manually download `walking_motion_processed.pkl` from [this Slack thread](https://cvmlp.slack.com/archives/C0460NTKM4G/p1678403985106999?thread_ts=1678402520.813389&cid=C0460NTKM4G) to `data/humanoids/humanoid_data/walking_motion_processed.pkl`
+    * Manually download `humanoids_skinned.zip` from [this link](https://drive.google.com/file/d/19gUvwaxJpd-Z6Djq8kmCpYotduwZvOfZ/view?usp=sharing) and uncompress to `data`
```
Comment (0mdc) [inline SIRO_README.md:17]: Could you add a semantic ID to the provided `ao_config.json` file? E.g. `"semantic_id": 100,`.

File: habitat-lab/habitat/articulated_agents/humanoids/kinematic_humanoid.py [hunk 2]
```diff
@@ -31,7 +31,7 @@ def _get_humanoid_params(self):
             wheel_mtr_vel_gain=None,
             wheel_mtr_max_impulse=None,
             ee_offset=[mn.Vector3(), mn.Vector3()],
-            ee_links=[14, 18],
+            ee_links=[20, 39],
```
Comment (0mdc) [inline habitat-lab/habitat/articulated_agents/humanoids/kinematic_humanoid.py:34]: Was this causing NANs when animating?

File: SIRO_README.md (hunk 1 above)
Comment (0mdc) [inline SIRO_README.md:17]: Could you add a semantic ID to the provided `ao_config.json` file? E.g. `"semantic_id": 100`,.

**Review by 0mdc — APPROVED** [review]
LGTM! Added some minor comments.

**Review by eundersander — APPROVED** [review]
I left a few comments but otherwise this looks good to me!

File: habitat-baselines/habitat_baselines/config/experiments_hab3/rearrange_fetch_human_planner.yaml [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+  environment:
+    max_episode_steps: 700
+
+  simulator:
+    kinematic_mode: True
+    ac_freq_ratio: 1
+    step_physics: False
+    agents:
+      agent_1:
+        articulated_agent_urdf: 'data/humanoids/humanoid_data/female2_0.urdf'
```
Comment (0mdc) [inline habitat-baselines/habitat_baselines/config/experiments_hab3/rearrange_fetch_human_planner.yaml:142]: The documentation mentions that we need to extract files in `data/` and some config files point there. Should this be `data/female2_0.urdf`?

File: habitat-baselines/habitat_baselines/config/experiments_hab3/socialnav_human_robot.yaml [hunk 4]
```diff
@@ -70,9 +70,9 @@ habitat:
   simulator:
     agents:
       agent_1:
-        articulated_agent_urdf: 'data/humanoids/humanoid_data/armatures/human_armature.urdf'
+        articulated_agent_urdf: 'data/humanoids/humanoid_data/female2_0.urdf'
```
Comment (0mdc) [inline habitat-baselines/habitat_baselines/config/experiments_hab3/socialnav_human_robot.yaml:73]: Same as above..

File: habitat-baselines/habitat_baselines/config/experiments_hab3/socialnav_human_robot_oracle.yaml [hunk 5]
```diff
@@ -94,9 +94,9 @@ habitat:
             width: 171
             hfov: 55
       agent_1:
-        articulated_agent_urdf: 'data/humanoids/humanoid_data/armatures/human_armature.urdf'
+        articulated_agent_urdf: 'data/humanoids/humanoid_data/female2_0.urdf'
```
Comment (0mdc) [inline habitat-baselines/habitat_baselines/config/experiments_hab3/socialnav_human_robot_oracle.yaml:97]: Same as above..

File: habitat-lab/habitat/config/benchmark/rearrange/rearrange_easy_human_and_spot.yaml [hunk 6]
```diff
@@ -35,7 +35,7 @@ habitat:
     agents:
       agent_0:
         radius: 0.3
-        articulated_agent_urdf: 'data/humanoids/humanoid_data/armatures/human_armature.urdf'
+        articulated_agent_urdf: 'data/humanoids/humanoid_data/female2_0.urdf'
```
Comment (0mdc) [inline habitat-lab/habitat/config/benchmark/rearrange/rearrange_easy_human_and_spot.yaml:38]: Same as above..

**Review by 0mdc — CHANGES_REQUESTED** [review]
Added some more comments.

File: habitat-lab/habitat/articulated_agents/humanoids/kinematic_humanoid.py (hunk 2 above)
Comment (xavierpuigf) [reply habitat-lab/habitat/articulated_agents/humanoids/kinematic_humanoid.py:35]: Sorry for the delay. Is this causing Nans?

File: habitat-lab/habitat/config/habitat/task/rearrange/pddl/multi_agent_social_nav_solved.yaml [hunk 7]
```diff
@@ -36,52 +36,22 @@ stage_goals:
...
-  - nav(robot_1, robot_0)
-  - nav(robot_1, robot_0)
-  - nav(robot_1, robot_0)
-  - nav(robot_1, robot_0)
-  - nav(robot_1, robot_0)
-  - nav(robot_1, robot_0)
-  - nav(robot_1, robot_0)
-  - nav(robot_1, robot_0)
-  - nav(robot_1, robot_0)
+  - nav_to_robot(robot_1, robot_0)
```
[the remark this answers is not in the document]
Comment (xavierpuigf) [reply habitat-lab/habitat/config/habitat/task/rearrange/pddl/multi_agent_social_nav_solved.yaml:47]: yes, but I will remove this. We have better solutions for this, and can move it to another branch called socialnav

**Pull request merged**